nginx/etc/nginx/nginx.conf.ctmpl

```diff
@@ -32,27 +40,66 @@ http {
         least_conn;
     }{{ end }}
 
+    # If we're listening on https, define an http listener that redirects everything to https
+    {{ if eq $ssl_ready "true" }}
+    server { 
+        server_name _;
+        listen      80;
 
-    server {
-        listen       80;
-        server_name  _;
-
-        root /usr/share/nginx/html;
-
+        # Respond to health requests defined in containerpilot.json
         location /nginx-health {
             stub_status;
             allow 127.0.0.1;
             deny all;
+            # Don't log these requests unless they fail
+            access_log /var/log/nginx/access.log  main if=$isError;
         }
 
+        location / {
+            return 301 https://$host$request_uri;
+        }
+    }
+    {{ end }}
+
+    server {
+        server_name _;
+        # Listen on port 80 unless we have certificates installed, then listen on 443
+        listen {{ if ne $ssl_ready "true" }}80{{ else }}443 ssl{{ end }};
+        {{ if eq $ssl_ready "true" }}
+        ssl_certificate /var/www/ssl/fullchain.pem;
+        ssl_certificate_key /var/www/ssl/privkey.pem;
+        ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+        ssl_prefer_server_ciphers on;
+        ssl_ciphers 'ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:AES128-GCM-SHA256:AES256-GCM-SHA384:AES128-SHA256:AES256-SHA256:AES128-SHA:AES256-SHA:AES:CAMELLIA:DES-CBC3-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!MD5:!PSK:!aECDH:!EDH-DSS-DES-CBC3-SHA:!EDH-RSA-DES-CBC3-SHA:!KRB5-DES-CBC3-SHA';
+        ssl_session_timeout 1d;
+        ssl_session_cache shared:SSL:50m;
+        ssl_stapling on;
+        ssl_stapling_verify on;
+        add_header Strict-Transport-Security max-age=15768000;
+        {{ end }}
+
         {{ if service "wordpress" }}
         rewrite ^/wp-admin/?(.*) /wordpress/wp-admin/$1;
-        
+
         location ^~ / {
             proxy_pass http://wordpress;
             proxy_set_header Host $http_host;
```

Comment:
I've been stuck on how I made this work before without adding `HTTP_X_FORWARDED_PROTO` and a conditional  in the `wp_config.php`. It's taken me a while, but the answer is this:

```
proxy_set_header Host $host:$server_port;
```